packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart

```diff
@@ -2,9 +2,1438 @@
+  /// null. If `quoted` is true, the `value` is enclosed
+  /// in double quotes.
+  factory DiagnosticsNode.stringProperty(String name, String value, {
+    bool showNull: true,
+    String header,
+    bool showName: true,
+    bool hidden: false,
+    bool quoted: false,
+  }) {
+    header ??= value;
+    if (quoted)
+      header = '"$header"';
+
+    return new _LeafDiagnosticsValue(
+      name: name,
+      header: header,
+      object: value,
+      showNull: showNull,
+      showName: showName,
+      hidden: hidden,
+    );
+  }
+
+  /// Property with a [bool] value.
+  ///
+  /// If `showNull` is true, the property is shown even if its `value` is null.
+  /// `header` overrides the normal description of the property value.
+  /// `showNull` indicates whether the property should still be shown if its
+  /// `value` is null. `showName indicates whether the property name should be
+  /// shown. `hidden` specifies whether the property should be hidden or shown.
+  /// `nullDescription` provides a description of the property value if it is
+  /// null.
+  factory DiagnosticsNode.boolProperty(String name, bool value, {
+    String header,
+    bool showNull: true,
+    bool showName: true,
+    bool hidden: false,
+    String nullDescription,
+  }) {
+    if (value == null)
+      header = nullDescription;
+
+    return new _LeafDiagnosticsValue(
+      name: name,
+      header: header ?? value.toString(),
+      object: value,
+      showNull: showNull,
+      hidden: hidden,
+      showName: showName,
+    );
+  }
+
+  /// A boolean flag property that is shown as `name` if value is true and is
+  /// true and hidden otherwise.
+  factory DiagnosticsNode.flagProperty(String name, bool value) {
+    return new _LeafDiagnosticsValue(
+      name: name,
+      header: value == true ? name : '',
+      object: value,
+      hidden: value != true,
+      showName: false);
+  }
+
+  /// An object property with a `comment` describing the value appended after
+  /// the description of the `value.
+  factory DiagnosticsNode.objectPropertyWithComment(String name, Object value, {
+    @required String comment,
+  }) {
+    return new _LeafDiagnosticsValue(
+      name: name,
+      object: value,
+      header: '$value ($comment)',
+    );
+  }
+
+  /// Creates a property with a [double] value.
+  ///
+  ///  If `showNull` is true, the property is shown even if its `value` is null.
+  /// `showNull` indicates whether the property should still be shown if its
+  /// `value` is null. `hidden` specifies whether the property should be hidden
+  /// or shown. `nullDescription` provides a description of the property value
+  /// if it is null.
+  /// If specified, exactly `fractionDigits` after the decimal point are shown.
+  /// If specified, `unit` describes the unit for the value (e.g. px).
+  factory DiagnosticsNode.doubleProperty(String name, double value, {
+    bool showNull: true,
+    int fractionDigits,
```

Comment:
Note that in the full code base, fractionDigits is only ever set to 1 or left unspecified.
Would you like to just force it to be 1 everywhere and remove the option? Happy to reduce the # of configuration levers. I expect that won't even change the golden output as Dart display of double values defaults to showing one digit past the decimal point even when not required.
Having the number of digits of precision shown change based on other properties on the node seems a bit spooky. Would be really odd to see an extra digit of precision appear just because I changed